deploy.sh

```diff
@@ -13,24 +13,26 @@ TERMAGGRS="top,significant,rare"
 APP_NAME="news-search-api"
 
 # Check if running as root
-if [[ $UID -ne 0 ]]; then
-    echo "This script must be run as root."
-    exit 1
-fi
+is_root() {
+    if [ $(whoami) != "root" ]; then
+        echo "This script must be run as root."
+        exit 1
+    fi
+}
+
+is_root
+echo "Running as root"
 
-LOGIN_USER=$(whoami)
+LOGIN_USER=$(who am i | awk '{ print $1 }')
 if [ "x$LOGIN_USER" = x ]; then
     # XXX fall back to whoami (look by uid)
     echo could not find login user 1>&2
     exit 1
 fi
 
 run_as_login_user() {
```

Comment:
I wrote this function to handle when the script is run as non-root so that it can do the right thing if the user is in group `docker` and don't need to sudo to root.

I don't see the point in invoking the `is_root` function here.